fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -375,6 +375,8 @@ public MvRefreshScheme copy() {
     @SerializedName(value = "baseTableInfos")
     private List<BaseTableInfo> baseTableInfos;
 
+    private transient List<TableType> baseTableTypes = Lists.newArrayList();
```

Comment:
Why not reuse the `baseTableInfos` to retrieve the table types ? 
Does it work as a cache ?

Reply:
What if table type changed ? e.g create a view named as `A`, drop `A`, create a table named as `A`

Reply:
fixed